src/domain/flow-system/src/entities/shared/schedule.rs

```diff
@@ -29,20 +33,63 @@ pub struct ScheduleTimeDelta {
 
 /////////////////////////////////////////////////////////////////////////////////////////
 
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct ScheduleCronExpression {
-    pub cron_expression: String,
+#[derive(Error, Debug)]
+pub enum ScheduleError {
+    #[error(transparent)]
+    InvalidCronExptression(#[from] InvalidCronExpressionError),
+
+    #[error(transparent)]
+    CronExpressionIterationExceed(#[from] CronExpressionIterationError),
+}
+
+#[derive(Error, Debug)]
+#[error("Cron expression {expression} is invalid")]
+pub struct InvalidCronExpressionError {
+    pub expression: String,
+}
+
+#[derive(Error, Debug)]
+#[error("Cron expression {expression} iteration has been exceed")]
+pub struct CronExpressionIterationError {
+    pub expression: String,
 }
 
 /////////////////////////////////////////////////////////////////////////////////////////
 
 impl Schedule {
-    pub fn next_activation_time(&self, now: DateTime<Utc>) -> DateTime<Utc> {
-        match self {
-            Schedule::TimeDelta(td) => now + td.every,
-            Schedule::CronExpression(_) => {
-                unimplemented!()
+    pub fn validate_cron_expression(
+        cron_expression: String,
+    ) -> Result<CronSchedule, ScheduleError> {
+        let schedule = match CronSchedule::from_str(&cron_expression) {
+            Err(_) => {
+                return Err(ScheduleError::InvalidCronExptression(
+                    InvalidCronExpressionError {
+                        expression: cron_expression.clone(),
+                    },
+                ));
             }
+            Ok(cron_schedule) => cron_schedule,
+        };
+        match schedule.upcoming(Utc).next() {
+            Some(_) => Ok(schedule),
+            None => Err(ScheduleError::CronExpressionIterationExceed(
+                CronExpressionIterationError {
+                    expression: cron_expression.clone(),
+                },
+            )),
+        }
+    }
+    pub fn next_activation_time(&self, now: DateTime<Utc>) -> Result<DateTime<Utc>, ScheduleError> {
```

Comment:
1) A blank line missing
2) I don't think we should mix error types between these 2 methods, since parsing error will not happen at this stage.